spokestack/asr/spokestack/keyword_recognizer.py

```diff
@@ -0,0 +1,195 @@
+"""
+This module contains the Spokestack KeywordRecognizer which identifies multiple keywords
+from an audio stream.
+"""
+import logging
+import os
+from typing import List
+
+import numpy as np  # type: ignore
+
+from spokestack.context import SpeechContext
+from spokestack.models.tensorflow import TFLiteModel
+from spokestack.wakeword.ring_buffer import RingBuffer
+
+
+_LOG = logging.getLogger(__name__)
+
+
+class KeywordRecognizer:
+    """Recognizes keywords in an audio stream.
+
+    Args:
+        classes (List[str]): Keyword labels
+        pre_emphasis (float): The value of the pre-emphasis filter
+        sample_rate (int): The number of audio samples per second of audio (kHz)
+        fft_window_type (str): The type of fft window. (only support for hann)
+        fft_hop_length (int): Audio sliding window for STFT calculation (ms)
+        model_dir (str): Path to the directory containing .tflite models
+        posterior_threshold (float): Probability threshold for detection
+    """
+
+    def __init__(
+        self,
+        classes: List[str],
+        pre_emphasis: float = 0.97,
+        sample_rate: int = 16000,
+        fft_window_type: str = "hann",
+        fft_hop_length: int = 10,
+        model_dir: str = "",
+        posterior_threshold: float = 0.5,
+        **kwargs
+    ) -> None:
+
+        self.classes = classes
```

Comment:
it would be good to validate the length of the class list against the output shape of the detector model